src/proto/streams/prioritize.rs

```diff
@@ -398,8 +398,18 @@ impl Prioritize {
             }
         }
 
-        // If data is buffered, then schedule the stream for execution
-        if stream.buffered_send_data > 0 {
+        // If data is buffered and the stream is not pending open, then
+        // schedule the stream for execution
+        //
+        // Why do we not push into pending_send when the stream is in pending_open?
+        //
+        // We allow users to call send_request() which schedules a stream to be pending_open
+        // if there is no room according to the concurrency limit (max_send_streams), and we
+        // also allow data to be buffered for send with send_data() if there is no capacity for
+        // the stream to send the data, which attempts to place the stream in pending_send.
+        // If the stream is not open, we don't want the stream to be
+        // scheduled for execution (pending_send)
```

Comment:
Really great comment!

Maybe also include that once the stream gets removed from pending_open it will be queued on `pending_send` (so the writes get scheduled at that point).

Reply:
Good point, added!